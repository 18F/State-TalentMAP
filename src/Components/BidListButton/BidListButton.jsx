import React from 'react';
import FontAwesome from 'react-fontawesome';

const BidListButton = () => (
  <button className="bid-list-button">
    <span className="bid-list-button-icon">
      <FontAwesome name="plus-circle" />
    </span>
    <span>Add to bid list</span>
  </button>
);

export default BidListButton;
