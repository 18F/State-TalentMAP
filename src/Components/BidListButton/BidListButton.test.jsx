import { shallow } from 'enzyme';
import sinon from 'sinon';
import React from 'react';
import toJSON from 'enzyme-to-json';
import BidListButton from './BidListButton';

describe('BidListButtonComponent', () => {
  const bidList = [{ id: 1 }];
  it('is defined', () => {
    const wrapper = shallow(
      <BidListButton
        id={1}
        toggleBidPosition={() => {}}
        compareArray={bidList}
        isLoading={false}
      />,
    );
    expect(wrapper).toBeDefined();
  });

  it('can handle text change when savedState is true', () => {
    const wrapper = shallow(
      <BidListButton
        id={1}
        toggleBidPosition={() => {}}
        compareArray={bidList}
        isLoading={false}
      />,
    );
    expect(wrapper.find('Add')).toBeDefined();
  });

  it('can handle text change when savedState is false', () => {
    const wrapper = shallow(
      <BidListButton
        id={1}
        toggleBidPosition={() => {}}
        compareArray={[{ id: 2 }]}
        isLoading={false}
      />,
    );
    expect(wrapper.find('Remove')).toBeDefined();
  });

  it('can call the toggleSaved function', () => {
    const spy = sinon.spy();
    const wrapper = shallow(
      <BidListButton
        id={1}
        toggleBidPosition={spy}
        compareArray={[{ id: 2 }]}
        isLoading={false}
      />,
    );
    wrapper.find('button').simulate('click');
    sinon.assert.calledOnce(spy);
  });

  it('matches snapshot', () => {
    const wrapper = shallow(
      <BidListButton
        id={1}
        toggleBidPosition={() => {}}
        compareArray={bidList}
        isLoading={false}
      />,
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
