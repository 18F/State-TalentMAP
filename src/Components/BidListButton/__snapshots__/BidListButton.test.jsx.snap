// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BidListButtonComponent matches snapshot 1`] = `
<button
  className="bid-list-button"
  onClick={[Function]}
  style={
    Object {
      "pointerEvents": "inherit",
    }
  }
>
  <span
    className="button-icon"
  >
    <FontAwesome
      name="plus-circle"
    />
  </span>
  <span>
    Add to Bid List
  </span>
</button>
`;
